src/sirinternal.c

```diff
@@ -1177,10 +1177,13 @@ pid_t _sir_gettid(void) {
     tid = (pid_t)pthread_self();
 #elif defined(__HAIKU__)
     tid = get_pthread_thread_id(pthread_self());
-#elif defined(__serenity__)
+#elif defined(__linux__) || defined(__serenity__)
+# if (defined(__GLIBC__) && (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 30)) || \
```

Comment:
repeated ifdef serenity

Reply:
purposeful?

Reply:
![image](https://github.com/aremmell/libsir/assets/61629094/4e22e410-4b35-44fc-85d3-00b9603b4cc1)

Yeah, needs to be that way, at least to avoid a separate elif block for just Serenity that also duplicates gettid:

If Linux or Serenity {
  If Serenity or Glibc2.30+ {
    tid = gettid();
  } else {
    tid = syscall();
  }
}